### PR TITLE
Ruby on Rails group project  🪄

### PR DESCRIPTION
**Pull request by ritikarawat220**
### What we have done as group in this project 🌳

- Follow the layout of the wireframes provided
- Login page and registration page
- Food list
- Recipes list
- Public recipe list
- Recipe details
    Etc......

### 🧠 Skills we gained 🌴

- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Understand Rails naming conventions.
- Use params from browser requests in a safe way.
- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
   Etc ...

### CODE REVIEW

**Comment by Shedrack-Sunday** [comment]
Hello @ritikarawat220 @jeanbuhendwa @cherelemma . Kindly add a README to your `dev` branch to get reviewed. 👍

File: app/views/recipe_foods/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <%= f.collection_select :food_id, @foods, :id, :name, prompt: "Select a food", class: 'form-control' %>
+  </div>
+  <br>
+  <div>
+    <%= f.label :quantity %>
+    <%= f.number_field :quantity, autofocus: true, autocomplete: "quantity", class: 'form-control' %>
+  </div>
+  <br>
+  <div>
+    <%= f.submit "Add Ingredient", class: 'btn btn-outline-dark btn-sm' %>
```
Comment (Shedrack-Sunday) [inline app/views/recipe_foods/new.html.erb:15]: - [ ] Good job on this feature 👏 , however, i tried to click the cancel button in an attempt that i changed my mind to add an ingredient, but the button does not work. Kindly look at the path and fix this bug to have a seamless webapp 👍 

### View:
image

File: app/views/recipes/shopping_list.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+     <div>
+       <p>Recipe: <%= link_to @recipe.name, recipe_path(@recipe) %></p>
+       <p>Inventory:</p>
+     </div>
+   </div>
+   <table class="table table-striped table-bordered">
+     <thead>
+       <tr>
+         <th scope="col">Food</th>
+         <th scope="col">Quantity</th>
```
Comment (Shedrack-Sunday) [inline app/views/recipes/shopping_list.html.erb:18]: - [ ] Nice work with the interface 👍 . However, the shopping list does not get rendered properly as you can see below. One of the causes is likely your pat route. Kindly debug and fix this to complement your brillant work so far 👍 

### View:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @ritikarawat220 , @jeanbuhendwa , and @cherelemma          👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with rails structured format✔️    
- Kudus with the validations ✔️ 
- Impressive and brillant deisgns ✔️ 

### Confirmed view brilliantly 👏 
index test
recipe list

  
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key point note I pointed out to meet the requirements before moving to the next project 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👏 You did a great fixing all the issues requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**